src/tools/ci/install_meson_and_rebase.sh

```diff
@@ -0,0 +1,66 @@
+#! /bin/sh
+
+set -e
+set -x
+
+case $CIRRUS_OS in
+    freebsd | linux | darwin | windows | mingw)
+    ;;
+    *)
+        echo "unsupported operating system ${CIRRUS_OS}"
+        exit 1
+    ;;
+esac
+
+# After repartition, hidden files are not copied. Copy them to
+# working dir
+if [ "$CIRRUS_OS" = 'freebsd' ]; then
+    cp -r $CIRRUS_WORKING_DIR.orig/.[^.]* $CIRRUS_WORKING_DIR/
+    chown -R postgres:postgres .[^.]*
+fi
+
+
+# install meson by using pip
+install_meson () {
+    case $CIRRUS_OS in
+        windows)
+            PYTHON=python
+        ;;
+        *)
+            PYTHON=python3
+        ;;
+    esac
+    ${PYTHON} -m pip install git+${MESON_REPO}@${MESON_BRANCH}
+}
+
+run_rebase_commands() {
+    git config user.email 'postgres-ci@example.com'
+    git config user.name 'Postgres CI'
+    # windows loses the executable bit, causing an unnecessary diff
+    git config core.filemode false
+    # windows changes file endings to crlf, causing test failures
+    git config core.autocrlf false
+    git reset --hard
+    git remote add default-postgres https://github.com/postgres/postgres.git
+    git fetch default-postgres master
+    git rebase --no-verify default-postgres/master
+}
+
+# Rebase current branch onto Postgres HEAD
+rebase_onto_postgres () {
+    echo "Rebasing onto: $(git show --no-patch --abbrev-commit  --pretty=oneline default-postgres/master)"
+    # freebsd and linux run commands as postgres user
```

Comment:
Is that actually required? It's true that they run configure / build as postgres, but that shouldn't be affected by source code permissions?